pymongo/collection.py

```diff
@@ -394,7 +395,7 @@ def database(self) -> "Database[_DocumentType]":
 
     def with_options(
         self,
-        codec_options: Optional[CodecOptions] = None,
+        codec_options: Optional["bson.CodecOptions[_DocumentTypeArg]"] = None,
```

Comment:
What caused the change to add "bson." here? We don't need to do the same for ReadConcern/WriteConcern/etcc..

Reply:
This was needed to satisfy sphinx:

```
/home/docs/checkouts/readthedocs.org/user_builds/pymongo/checkouts/1139/pymongo/collection.py:docstring of pymongo.collection.Collection.with_options:: WARNING: more than one target found for cross-reference 'CodecOptions': bson.codec_options.CodecOptions, bson.CodecOptions
/home/docs/checkouts/readthedocs.org/user_builds/pymongo/checkouts/1139/pymongo/database.py:docstring of pymongo.database.Database:: WARNING: more than one target found for cross-reference 'CodecOptions': bson.codec_options.CodecOptions, bson.CodecOptions
/home/docs/checkouts/readthedocs.org/user_builds/pymongo/checkouts/1139/pymongo/database.py:docstring of pymongo.database.Database.create_collection:: WARNING: more than one target found for cross-reference 'CodecOptions': bson.codec_options.CodecOptions, bson.CodecOptions
/home/docs/checkouts/readthedocs.org/user_builds/pymongo/checkouts/1139/pymongo/database.py:docstring of pymongo.database.Database.get_collection:: WARNING: more than one target found for cross-reference 'CodecOptions': bson.codec_options.CodecOptions, bson.CodecOptions
/home/docs/checkouts/readthedocs.org/user_builds/pymongo/checkouts/1139/pymongo/database.py:docstring of pymongo.database.Database.with_options:: WARNING: more than one target found for cross-reference 'CodecOptions': bson.codec_options.CodecOptions, bson.CodecOptions
/home/docs/checkouts/readthedocs.org/user_builds/pymongo/checkouts/1139/pymongo/mongo_client.py:docstring of pymongo.mongo_client.MongoClient.get_default_database:: WARNING: more than one target found for cross-reference 'CodecOptions': bson.codec_options.CodecOptions, bson.CodecOptions
/home/docs/checkouts/readthedocs.org/user_builds/pymongo/checkouts/1139/pymongo/mongo_client.py:docstring of pymongo.mongo_client.MongoClient.get_database:: WARNING: more than one target found for cross-reference 'CodecOptions': bson.codec_options.CodecOptions, bson.CodecOptions
```

Reply:
That's odd. I don't see those warning locally: 
```
 python -m sphinx -b html doc doc/_build/4.4.0.dev0
Running Sphinx v4.5.0
loading pickled environment... failed
failed: build environment version not current
loading intersphinx inventory from https://www.gevent.org/objects.inv...
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 83 source files that are out of date
updating environment: [new config] 83 added, 0 changed, 0 removed
reading sources... [100%] tutorial                                                                                                                      
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] tutorial                                                                                                                       
generating indices... genindex py-modindex done
writing additional pages... search done
copying images... [100%] static/periodic-executor-refs.png                                                                                              
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in doc/_build/4.4.0.dev0.
```

It would be great to avoid changing the type signature just because of a sphinx issue (bug?) if possible.

Reply:
https://github.com/sphinx-doc/sphinx/issues/10400

Reply:
We needed to switch to strings because on Python 3.7 we were getting: `TypeError: 'type' object is not subscriptable`